docs/advanced/runtime/docs/mono-error.md

```diff
@@ -0,0 +1,126 @@
+---
+title: Error handling and MonoError
+---
+
+MonoError
+=========
+
+MonoError is the latest attempt at cleaning up and sanitizing error handling in the runtime.
+This document highlights some of the design goals and decisions, the implementation and the migration strategy.
+
+## Design goals
+
+- Replace the majority of the adhoc error handling subsystems present today in the runtime. Each one is broken
+is a subtle way, has slightly different semantics and error conversion between them is spot, at best.
+
+- Map well to the final destination of all runtime errors: managed exceptions. This includes being compatible
+with .net when it comes to the kind of exception produced by a given error condition.
+
+- Be explicit, lack any magic. The loader-error setup does control flow happens in the background through a TLS variable,
+which made it very brittle and error prone.
+
+- Explicit and multiple error scopes. Make it possible to have multiple error scopes and make them explicit. We need to
+support nested scopes during type loading, even if reporting is flat.
+
+- Be as simple as possible. Error handling is the hardest part of the runtime to test so it must be simple. Which means
+complex error reporting, such as chaining, is out of question.
+
+Current implementation
+======================
+
+The current implementation exists in mono-error.h and mono-error-internals.h. The split is so API users can consume errors, but they
+are not supported to be able to produce them - such use case has yet to arise.
+
+### Writing a function that produces errors
+
+```
```

Comment:
You can add `c` after the backticks and we'll get a nice syntax highlighting on the website.